server/src/main/java/io/crate/expression/scalar/DateBinFunction.java

```diff
@@ -107,7 +110,15 @@ public final Long evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input
             return null;
         }
 
-        return getBinnedTimestamp(((Period) interval).toStandardDuration().getMillis(), (long) timestamp, (long) origin);
+        Period p = (Period) interval;
+        checkMonthsAndYears(p);
+        return getBinnedTimestamp(p.toStandardDuration().getMillis(), (long) timestamp, (long) origin);
+    }
+
+    private static void checkMonthsAndYears(Period p) {
+        if (p.getMonths() != 0 || p.getYears() != 0) {
```

Comment:
Can we add a javadoc, saying that it's similar to Joda Time's `Period.checkYearsAndMonths` but with more relevant for us error message?

Reply:
Sure! thx!